packages/identity-secp256k1/src/pem.ts

```diff
@@ -0,0 +1,42 @@
+import { concat, uint8ToBuf } from '@dfinity/agent';
+
+const HEADER = `-----BEGIN`;
+const FOOTER = `-----END`;
+const ECHEADER = `-----BEGIN EC PRIVATE KEY`;
+
+/**
+ * Parse a PEM-encoded key into an ArrayBuffer
+ * @param pem - the PEM-encoded key
+ * @returns secret key as an ArrayBuffer
+ */
+export function pemToSecretKey(pem: string): ArrayBuffer {
+  const lines = pem.trim().split('\n');
+  const header = lines[0].trim();
+  const footer = lines[lines.length - 1].trim();
+  if (lines.length < 3) {
+    throw new Error('Invalid PEM format');
+  }
+  if (!header.startsWith(HEADER)) {
+    throw new Error('Invalid PEM header');
+  }
+  if (!footer.startsWith(FOOTER)) {
+    throw new Error('Invalid PEM footer');
+  }
+  const base64Data = lines.slice(1, -1).join('').replace(/\r?\n/g, '');
+  const rawKey = Uint8Array.from(atob(base64Data), c => c.charCodeAt(0));
+
+  if (pem.startsWith(ECHEADER)) {
+    if (rawKey.length !== 118) {
+      throw new Error(`Invalid key length ${rawKey.length}. Expected 118 bytes.`);
+    } else {
+      return rawKey.slice(7, 39);
+    }
+  }
+  if (rawKey.length != 85) {
+    throw new Error(`Invalid key length ${rawKey.length}. Expected 85 bytes.`);
+  } else {
+    return concat(rawKey.slice(16, 48), rawKey.slice(53, 85));
```

Comment:
I have no idea what is happening here :)

It seems like this is assuming that if the PEM header is not "EC PRIVATE KEY" then it is some other specific unnamed format, and that performing this operation where we remove (?) 5 bytes from the middle is appropriate.

At the very least this deserves a comment regarding what data is being removed, and probably a check that the header is the expected one.

Reply:
Oh I see from the tests I'm guessing this is `PRIVATE KEY`

A comment on the format and which bits are being extracted seems warranted.

Also should point out that `PRIVATE KEY` is in practice used for many different kinds of key, so if you put an RSA private key, or even just an encrypted ECDSA key, this function will just return some junk. If supporting this format is necessary you should probably at a minimum check the leading bytes, which will contain a key identifier

```
    0:d=0  hl=3 l= 132 cons: SEQUENCE          
    3:d=1  hl=2 l=   1 prim: INTEGER           :00
    6:d=1  hl=2 l=  16 cons: SEQUENCE          
    8:d=2  hl=2 l=   7 prim: OBJECT            :id-ecPublicKey
   17:d=2  hl=2 l=   5 prim: OBJECT            :secp256k1
   24:d=1  hl=2 l= 109 prim: OCTET STRING      [HEX DUMP]:....
```